hyper/http11/connection.py

```diff
@@ -93,11 +105,16 @@ def connect(self):
         :returns: Nothing.
         """
         if self._sock is None:
-            sock = socket.create_connection((self.host, self.port), 5)
+            if not self.proxy_host:
+                host = self.host
+            else:
+                port = self.proxy_host
+                
+            sock = socket.create_connection((host, port), 5)
             proto = None
 
             if self.secure:
-                sock, proto = wrap_socket(sock, self.host, self.ssl_context)
+                sock, proto = wrap_socket(sock, host, self.ssl_context)
```

Comment:
I did overlook this and forget about connect.  Will go ahead and use assert like you suggested.